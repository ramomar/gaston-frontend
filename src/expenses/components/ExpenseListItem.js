import React from 'react';
import { Text, Box } from 'grommet';
import { DateTime } from 'luxon';

function ExpenseListItem({ expense, toExpenseReviewScreen }) {
  const {
    note,
    amount,
    date
  } = expense;

  return (
    <Box
      height='xxsmall'
      margin={{ bottom: 'medium' }}
      pad={{ vertical: 'large' }}
      direction='row'
      align='center'
      justify='between'
      onClick={() => toExpenseReviewScreen()}>
      <Box basis='2/4'>
        <Text truncate>{note}</Text>
      </Box>
      <Box align='end' basis='2/4'>
        <Text size='small' weight='bold'>-${amount.toFixed(2)}</Text>
        <Text size='small'>{date.toLocaleString(DateTime.TIME_24_SIMPLE)}</Text>
      </Box>
    </Box>
  );
}

export default ExpenseListItem;
