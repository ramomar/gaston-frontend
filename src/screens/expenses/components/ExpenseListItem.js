import React from 'react';
import { Text, Box } from 'grommet';
import { useHistory } from 'react-router-dom';
import { DateTime } from 'luxon';

function ExpenseListItem({ expense }) {
  const { push } = useHistory();

  const {
    id,
    note,
    amount,
    date
  } = expense;

  const toReviewScreen = () => push(`/expenses/${id}/review`, {
    expense: {
      id,
      note,
      amount,
      date: date.toISO()
    },
    fromList: true
  });

  return (
    <Box
      height='xxsmall'
      margin={{ bottom: 'medium' }}
      pad={{ vertical: 'large' }}
      direction='row'
      align='center'
      justify='between'
      onClick={toReviewScreen}>
      <Box basis='2/4'>
        <Text truncate>{note}</Text>
      </Box>
      <Box align='end' basis='2/4'>
        <Text size='small' weight='bold'>-${amount.toFixed(2)}</Text>
        <Text size='small'>{date.toLocaleString(DateTime.TIME_24_SIMPLE)}</Text>
      </Box>
    </Box>
  );
}

export default ExpenseListItem;
