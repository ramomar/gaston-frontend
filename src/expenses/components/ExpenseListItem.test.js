import React from 'react';
import renderer from 'react-test-renderer';
import ExpenseListItem from './ExpenseListItem';
import { DateTime } from 'luxon';

it('renders correctly', () => {
  const expense = {
    'id': '0007182d-54cb-42b7-88fc-bbaba51db198',
    'amount': 150,
    'date': DateTime.fromISO('2017-03-19T05:29:02.700Z'),
    'note': 'Cena'
  };

  const component =
    <ExpenseListItem
      expense={expense}
      toExpenseReviewScreen={() => null} />;

  const tree = renderer.create(component).toJSON();

  expect(tree).toMatchSnapshot();
});
