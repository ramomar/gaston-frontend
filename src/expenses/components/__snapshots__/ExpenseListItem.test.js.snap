// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="StyledBox-sc-13pk1d4-0 vtIgb"
  onBlur={[Function]}
  onClick={[Function]}
  onFocus={[Function]}
  onKeyDown={[Function]}
  tabIndex={0}
>
  <div
    className="StyledBox-sc-13pk1d4-0 cHMbYi"
  >
    <span
      className="StyledText-sc-1sadyjn-0 jOAXsc"
    >
      Cena
    </span>
  </div>
  <div
    className="StyledBox-sc-13pk1d4-0 gIKgZq"
  >
    <span
      className="StyledText-sc-1sadyjn-0 isVaoK"
      size="small"
    >
      -$
      150.00
    </span>
    <span
      className="StyledText-sc-1sadyjn-0 gNZQxf"
      size="small"
    >
      23:29
    </span>
  </div>
</div>
`;
